### PR TITLE
[Feature] support dbt macros

### PR DESCRIPTION
**Pull request by qrtt1**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**

feature added

**What this PR does / why we need it**:

leverage dbt packages 

**Which issue(s) this PR fixes**:

#31 

**Special notes for your reviewer**:

demos
----

```yaml
{% set a_query %}
select * from {{ ref("cloud_user_last_seen") }} where id>100
{% endset %}

{% set b_query %}
select * from {{ ref("cloud_user_last_seen") }} where id<100
{% endset %}

{{ audit_helper.compare_queries(
    a_query=a_query,
    b_query=b_query,
    primary_key="id"
) }}
```

image

### CODE REVIEW

**Comment by popcornylu** [comment]
@qrtt1 thanks

Two recommendataion.
1. Use the dbt compile by default. I think there is no reason to use the old one. (i.e. Remove the feature flg `recce_sql_engine` )
2. Can we just put the code in `dbt.py`? Or move the dbt to the package `recce.dbt`?
3. Use `audit_helper` seems not suitable for current query diff, could you change to use [dbt_profiler.get_profile](https://github.com/data-mie/dbt-profiler/tree/0.8.1/#get_profile-source)

**Comment by qrtt1** [comment]
@popcornylu old implementation removed

image

**Comment by popcornylu** [comment]
@qrtt1  it seems that the query for base always run the result of the current environment.

Could you provide a screenshot that show there is a difference found?

File: recce/dbt.py [hunk 1]
```diff
@@ -111,37 +175,24 @@ def find_source_by_name(self, source_name, table_name, base=False) -> Optional[S
...
-
-            relation = self.adapter.Relation.create_from(self.project, source)
-            return str(relation)
-
-        template = Template(sql_template)
-        sql = template.render(ref=ref, source=source)
-
         adapter = self.adapter
         with adapter.connection_named('test'):
+            sql = self.generate_sql(sql_template)
```
Comment (popcornylu) [inline recce/dbt.py]: base argument is missing?

**Review by popcornylu — APPROVED** [review]

**Comment by qrtt1** [comment]
Diff of profiler :D

```sql
{{ dbt_profiler.get_profile(relation=ref("cloud_user_last_seen")) }}
```
```patch
diff --git a/models/marts/cloud_user_last_seen.sql b/models/marts/cloud_user_last_seen.sql
index a03db61..f15b4e8 100644
--- a/models/marts/cloud_user_last_seen.sql
+++ b/models/marts/cloud_user_last_seen.sql
@@ -9,6 +9,7 @@ WITH ReportDate AS (
         MAX(created_at) AS last_seen,
         DATEDIFF(day, MIN(created_at), MAX(created_at)) AS active_days
     FROM {{ ref('stg_cloud__audit_logs') }}
+    WHERE user_id > 100
     GROUP BY user_id
 )
 


```

image

**Pull request merged**